src/timely-util/src/operator.rs

```diff
@@ -390,6 +401,27 @@ where
             }
         })
     }
+
+    fn with_token(&self, token: Weak<()>) -> Stream<G, D1> {
+        let mut builder = OperatorBuilderRc::new("WithToken".to_owned(), self.scope());
+
+        let mut input = builder.new_input(self, Pipeline);
+        let (mut output, stream) = builder.new_output();
+
+        let mut vector = Default::default();
+        builder.build(move |_capability| {
+            move |_frontier| {
+                let mut output = output.activate();
```

Comment:
I suppose this could be expressed directly as a `unary` operator, thus avoiding the Timely boilerplate?

Reply:
Good catch!